dashboard/app/controllers/feature_mode_controller.rb

```diff
@@ -0,0 +1,64 @@
+# A controller that allows admins to update the current feature mode using the
+# FeatureModeManager. Please see that class for an explanation of the various
+# feature modes.
+
+class FeatureModeController < ApplicationController
+  before_filter :authenticate_user!
+
+  # Max time in seconds for settings updates to take effect.
+  MAX_UPDATE_TIME = 30
+
+  PLEASE_WAIT_MESSAGE = "Updating feature mode. Please wait #{MAX_UPDATE_TIME} seconds for changes to take effect."
+
+  # Shows the current or pending feature mode.
+  def show
+    authorize! :read, :reports
+    @current_mode = FeatureModeManager.get_mode(Gatekeeper, DCDO, ScriptConfig.cached_scripts)
+    @pending_mode = pending_mode
+    # If a mode update is still pending, show that mode and display a notice.
+    if @pending_mode && @current_mode != @pending_mode
+      @mode = @pending_mode
+      flash[:notice] = PLEASE_WAIT_MESSAGE
+    # Otherwise show the mode determined from the gatekeeper settings (if any)
+    elsif @current_mode
+      @mode =  @current_mode
+    else
+      @mode = 'custom'
+      flash[:alert] = 'The current feature flags do not match any of the pre-defined modes.'
+    end
+  end
+
+  # Updates the feature mode based on params[:mode].
+  def update
+    mode = params[:mode]
```

Comment:
Ruby-magic question: Since you've omitted `authorize! :read, :reports` here, does that mean any signed-in user could `POST` to /admin/feature_mode and change our current mode?  Or is there some awesome admin-check stuff going on elsewhere?